arangod/Aql/Condition.cpp

```diff
@@ -1221,7 +1220,7 @@ void Condition::optimize(ExecutionPlan* plan, bool multivalued) {
                   _ast->createNodeBinaryOperator(NODE_TYPE_OPERATOR_BINARY_IN,
                                                  leftNode->getMemberUnchecked(0),
                                                  mergeInOperations(trx, leftNode, rightNode));
-              andNode->removeMemberUnchecked(rightPos);
+              andNode->removeMemberUncheckedUnordered(rightPos);
```

Comment:
I am a bit concerned that this change will unintentionally break something.
The code around this location (above) seems to be order-dependent, and I don't know what will happen if we swap items in the `members` vector. Can we exclude this location when making the changes?
The same applies for the following few occurrences.
If you think this is a big performance hit for arangosearch, can you make the change conditional, i.e.
```
if (multivalued) {
  node->removeMemberUncheckedUnordered(...);
} else {
  node->removeMemberUnchecked(...);
}
```